artemis-jakarta-client/pom.xml

```diff
@@ -0,0 +1,95 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.activemq</groupId>
+        <artifactId>artemis-pom</artifactId>
+        <version>2.17.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>artemis-jakarta-client</artifactId>
+    <packaging>jar</packaging>
+    <name>ActiveMQ Artemis Jakarta Messaging Client</name>
+
+    <properties>
+        <activemq.basedir>${project.basedir}/..</activemq.basedir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-jms-client</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+            <!-- needed for the maven reactor -->
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-core-client</artifactId>
+            <version>${project.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-commons</artifactId>
+            <version>${project.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-selector</artifactId>
+            <version>${project.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.jms</groupId>
+            <artifactId>jakarta.jms-api</artifactId>
+            <scope>runtime</scope>
```

Comment:
I think the API should be compile scoped. A non-transformed implementation wouldn't get away without it being available at compile time. Using this module as a dep with it only specified at runtime scope means you must provide the API yourself at compile time (which isn't actually bad in itself, but still likely to be unexpected/confusing to many and another difference to artemis-jms-client).

Presumably the transformation does have it at compile time already, due to using the dependency-managed 'provided' scope from the parent making it available? Perhaps that should also be changed to not say provided?

Reply:
The transform does bytecode transformation directly, I'm not sure the API is required as I don't see it in the batavia nor in the eclipse transformer poms. I think we could do without. I don't think there is any compilation here.

Reply:
I would still list it as a compile scoped, for the knock on effect, and taking into account that it will be compile scoped in future if say the module stopped being transformed from the artemis-jms-client module. As I say, depending on artemis-jms-client gets you the required JMS API. It seems reasonable using artemis-jakarta-client gets you the required Jakarta Messaging API.